File: PDOO/NapakalakiRuby/lib/Monster.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
  class Monster
    attr_reader :name, :combatLevel, :badConsequence, :buenrollo

    private_class_method :new

    def initialize(n,cl,ml,br,lcacp)
      @name        = n
      @combatLevel = cl
      @badConsequence = ml
      @buenrollo   = br
      @levelChangeAgainstCultistPlayer = lcacp
    end

    def self.newMonsterLevelChangeAgainstCultistPlayer(n,i,bc,p,ic)
      new(n,i,bc,p,ic)
    end

    def self.newMonster(n,i,bc,p)
      new(n,i,bc,p,0)
    end

    def to_s

         "\nSoy #{@name} \nNivel: #{@combatLevel} \nMi mal rollo es: " + @badConsequence.to_s + "\nMi buen rollo es: " + @buenrollo.to_s

    end

    public
    def getCombatLevel
      @combatLevel
    end

    def getLevelsGained

      @buenrollo.levels

    end

    def getTreasuresGained

      @buenrollo.treasures

    end

    def getCombatLevelAgainstCultistPlayer
      @combatLevel + @levelChangeAgainstCultisPlayer
    end
  end
end

File: PDOO/NapakalakiJava/src/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author adritake
 */
public class Monster {
    private String name; 
    private int combatLevel;
    private Prize buenrollo;
    private BadConsequence malrollo;
    private int levelChangeAgainstCultistPlayer;
            
    public Monster(String name, int level, BadConsequence bc, Prize price){
        this.name = name;
        this.combatLevel = level;
        this.malrollo = bc;
        this.buenrollo = price;
        this.levelChangeAgainstCultistPlayer = 0;
    }
    
    public Monster(String name, int level, BadConsequence bc, Prize price ,int ic){
        this.name = name;
        this.combatLevel = level;
        this.malrollo = bc;
        this.buenrollo = price;
        this.levelChangeAgainstCultistPlayer = ic;
    }
    
    public Monster(){
        
        
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        
        return combatLevel + levelChangeAgainstCultistPlayer;
        
    }
    public String getName(){
        return name; 
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public BadConsequence getBadConsequence(){
        return malrollo;
    }
    
    public Prize getBuenRollo(){
        return buenrollo;
    }
    
    public int getLevelsGained(){
        
        return buenrollo.getLevel();
        
    }
    
    public int getTreasuresGained(){
        
        return buenrollo.getTreasures();
        
    }
    
    public String toString(){
        return "Nombre: " + name + "\nLevel: " + Integer.toString(combatLevel)
                + "\nBuen rollo: " + buenrollo.toString() + "\nMal rollo: " 
                + malrollo.toString();
                        
    }
    
    
    
  
}
